File: bit/maximum_and_value.cpp
//{ Driver Code Starts
// Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
 public:
  // Function for finding maximum AND value.
  int maxAND(int arr[], int N) {
    // Your code here
    // 1e5 is less than 17 digits in binary
    int ans = 0;
    vector<int> a(arr, arr + N);
    for (int k = 17; k >= 0; --k) {
      vector<int> v;
      for (int i = 0; i < a.size(); ++i) {
        if (a[i] & (1 << k)) {
          v.push_back(a[i]);
        }
      }
      // cout << k << ' ' << v.size() << endl;
      if (v.size() >= 2) {
        ans = max(ans, v[0] & v[1]);
        swap(a, v);
      }
    }
    return ans;
  }
};

//{ Driver Code Starts.

// Driver function
int main() {
  int t;
  cin >> t;  // testcases
  while (t--) {
    int n;
    cin >> n;  // input n
    int arr[n + 5], i;

    // inserting elements
    for (i = 0; i < n; i++) cin >> arr[i];
    Solution obj;
    // calling maxAND() function
    cout << obj.maxAND(arr, n) << endl;
  }
  return 0;
}
// } Driver Code Ends

File: bit/maximum_and_value.py
#User function Template for python3
class Solution:

	def maxAND (self, arr,  n) :
		# Complete the function
		ans = 0
        for i in range(18, -1, -1):
            temp = []
            for a in arr:
                if a & (1 << i):
                    temp.append(a)
            
            if len(temp) == 2:
                return temp[0] & temp[1]
            elif len(temp) > 2:
                arr = temp
                ans += (1 << i)
        
        return ans
                

#{ 
 # Driver Code Starts
#Initial Template for Python 3


   
for _ in range(0,int(input())):
    
    n = int(input())
    arr = list(map(int,input().strip().split()))
    ob = Solution()
    print(ob.maxAND(arr, n))
# } Driver Code Ends
